server/http_test.go

```diff
@@ -842,3 +949,593 @@ func TestGetSubscriptionsForChannel(t *testing.T) {
+		},
+		"Not Authorized": {
+			subscriptionTemplate:       "{}",
+			expectedStatusCode: http.StatusUnauthorized,
+			skipAuthorize:      true,
+		},
+		"Initial Subscription Template": {
+			subscriptionTemplate:       `{"instance_id": "jiraurl1", "name": "some name", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": ["10001"]}}`,
+			expectedStatusCode: http.StatusOK,
+			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{
+				{
+					Filters: SubscriptionFilters{
+						Events:     NewStringSet("jira:issue_created"),
+						Projects:   NewStringSet("myproject"),
+						IssueTypes: NewStringSet("10001"),
+					},
+				},
+			}, nil, t),
+		},
+		"Initial Subscription Template, GetProject mocked error": {
+			subscriptionTemplate:       fmt.Sprintf(`{"instance_id": "jiraurl1", "name": "some name", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "filters": {"events": ["jira:issue_created"], "projects": ["%s"], "issue_types": ["10001"]}}`, nonExistantProjectKey),
+			expectedStatusCode: http.StatusInternalServerError,
+			apiCalls:           hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Initial Subscription Template, empty name provided": {
+			subscriptionTemplate:       `{"instance_id": "jiraurl1", "name": "", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": ["10001"]}}`,
+			expectedStatusCode: http.StatusInternalServerError,
+			apiCalls:           hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Initial Subscription Template, long name provided": {
+			subscriptionTemplate:       `{"instance_id": "jiraurl1", "name": "` + TestDataLongSubscriptionName + `", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": ["10001"]}}`,
+			expectedStatusCode: http.StatusInternalServerError,
+			apiCalls:           hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Initial Subscription Template, no project provided": {
+			subscriptionTemplate:       `{"instance_id": "jiraurl1", "name": "somename", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "filters": {"events": ["jira:issue_created"], "projects": [], "issue_types": ["10001"]}}`,
+			expectedStatusCode: http.StatusInternalServerError,
+			apiCalls:           hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Initial Subscription Template, no issue types provided": {
+			subscriptionTemplate:       `{"instance_id": "jiraurl1", "name": "somename", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "filters": {"events": ["jira:issue_created"], "projects": ["myproject"], "issue_types": []}}`,
+			expectedStatusCode: http.StatusInternalServerError,
+			apiCalls:           hasSubscriptionTemplates([]SubscriptionTemplate{}, t),
+		},
+		"Adding to existing with other channel": {
```

Comment:
`Adding existing Subscription Template to other channels` I think the description should be this